src/main/java/mekanism/common/item/interfaces/IJetpackItem.java

```diff
@@ -0,0 +1,157 @@
+
+    JetpackMode getJetpackMode(ItemStack stack);
+
+    void useJetpackFuel(ItemStack stack);
+
+    enum JetpackMode implements IIncrementalEnum<JetpackMode>, IHasTextComponent {
+        NORMAL(MekanismLang.JETPACK_NORMAL, EnumColor.DARK_GREEN, MekanismUtils.getResource(ResourceType.GUI_HUD, "jetpack_normal.png")),
+        HOVER(MekanismLang.JETPACK_HOVER, EnumColor.DARK_AQUA, MekanismUtils.getResource(ResourceType.GUI_HUD, "jetpack_hover.png")),
+        DISABLED(MekanismLang.JETPACK_DISABLED, EnumColor.DARK_RED, MekanismUtils.getResource(ResourceType.GUI_HUD, "jetpack_off.png"));
+
+        private static final JetpackMode[] MODES = values();
+        private final ILangEntry langEntry;
+        private final EnumColor color;
+        private final ResourceLocation hudIcon;
+
+        JetpackMode(ILangEntry langEntry, EnumColor color, ResourceLocation hudIcon) {
+            this.langEntry = langEntry;
+            this.color = color;
+            this.hudIcon = hudIcon;
+        }
+
+        @Override
+        public Component getTextComponent() {
+            return langEntry.translateColored(color);
+        }
+
+        @Nonnull
+        @Override
+        public JetpackMode byIndex(int index) {
+            return byIndexStatic(index);
+        }
+
+        public ResourceLocation getHUDIcon() {
+            return hudIcon;
+        }
+
+        public static JetpackMode byIndexStatic(int index) {
+            return MathUtils.getByIndexMod(MODES, index);
+        }
+    }
+
+    /**
+     * Gets the first found active jetpack from an entity, if one is worn.
+     * <br>
+     * If Curios is loaded, the curio slots will be checked as well.
+     *
+     * @param entity the entity on which to look for the jetpack
+     *
+     * @return the jetpack stack if present, otherwise an empty stack
+     */
+    @Nonnull
+    static ItemStack getActiveJetpack(LivingEntity entity) {
+        return getJetpack(entity, stack -> stack.getItem() instanceof IJetpackItem jetpackItem && jetpackItem.canUseJetpack(stack));
+    }
+
+    /**
+     * Gets the first found jetpack from an entity, if one is worn. Purpose of this is to get the correct jetpack mode to use.
+     * <br>
+     * If Curios is loaded, the curio slots will be checked as well.
+     *
+     * @param entity the entity on which to look for the jetpack
+     *
+     * @return the jetpack stack if present, otherwise an empty stack
+     */
+    @Nonnull
+    static ItemStack getPrimaryJetpack(LivingEntity entity) {
+        return getJetpack(entity, stack -> stack.getItem() instanceof IJetpackItem);
+    }
+
+    private static ItemStack getJetpack(LivingEntity entity, Predicate<ItemStack> matcher) {
+        ItemStack chest = entity.getItemBySlot(EquipmentSlot.CHEST);
+        if (matcher.test(chest)) {
+            return chest;
+        } else if (Mekanism.hooks.CuriosLoaded) {
+            return CuriosIntegration.findFirstCurio(entity, s -> matcher.test(s));
```

Comment:
You should be able to just pass matcher directly as a parameter to `findFirstcurio` or is there something I am overlooking?

Reply:
nope, just missed it